### ISSUE TITLE
Incorrect interface event data, cause not clear

### ISSUE THREAD

**Issue by eduran84** [Yousei9/Logistic-Train-Network#167]
This started with an error caused by LTNT: https://forums.factorio.com/viewtopic.php?f=214&t=64842&start=60#p418220.
I've tracked the issue down to LTN sending incorrect event data. Annotated log showing the issue:
[factorio-current.log](https://github.com/Yousei9/Logistic-Train-Network/files/3025302/factorio-current.log)
The most pertinent part is this:

```
2317.880 Script @__LTN_Tracker__/ltnt/logger.lua:180: [LTNT][INFO]<on_pickup_completed> event: {
  delivery = {
    force = {},
    train = {[...]}
    started = 61047,
    from = "[img=item/electronic-components][img=item/logistic-chest-passive-provider] Electronic Components",
    to = "[img=item/electronic-components][img=item/logistic-chest-passive-provider] Electronic Components",
    networkID = 4294967295,
    shipment = {
      ["[img=item/electronic-components][img=item/logistic-chest-passive-provider] Electronic Components"] = 1
    },
    pickupDone = true
  },
  trainID = 4,
  name = 148,
  tick = 61629,
  mod_name = "LogisticTrainNetwork"
}
```
from, to and item name are all identical. Weirdly, LTN logging shows no issue and correctly named items. But the event data is messed up on arrival. I have no clue how that is possible. Probably something to do with icons as part of station names?

**To Reproduce**
Load [save2.zip](https://github.com/Yousei9/Logistic-Train-Network/files/3025306/save2.zip), sync mods, run for ~30 seconds.

**LTN version**
LTN 1.10.4, LTNT 0.9.7, Factorio 0.17.23

**Comment by eduran84** [comment]
Some further info: everything works if none of the stations include rich text in their names. If the names do include rich text, both to and from are identical in event data. But for most items the item name is correct anyway (i.e. not causing an error in LTNT, just messing with the displayed history).
I can't make sense of this. It might be a vanilla bug.

**Comment by Yousei9** [comment]
Here's a save stripped down to LTN:
[crash-test12a.zip](https://github.com/Yousei9/Logistic-Train-Network/files/3025425/crash-test12a.zip)

This bug is most peculiar. 
I have no idea why these rich texts mess up event data. I guarantee the string LTN writes to the event is correct and the same my internal debug log reports.

**Comment by Yousei9** [comment]
I reached Klonan on discord, he seems to be working on a way to reproduce and hopefully either fix it in Factorio or tell me what I did wrong.

**Comment by eduran84** [comment]
I've managed to create a minimal working example without LTN, so it must be a base game bug.
[receiver_0.1.0.zip](https://github.com/Yousei9/Logistic-Train-Network/files/3026353/receiver_0.1.0.zip)
[sender_0.1.0.zip](https://github.com/Yousei9/Logistic-Train-Network/files/3026354/sender_0.1.0.zip)


```
639.717 Script @__sender__/control.lua:12: Data, before event raised:
{
  string1 = "[img=item/express-transport-belt] String A",
  string2 = "[img=item/express-transport-belt] String B"
}
 639.718 Script @__receiver__/control.lua:3: Event received:
{
  data = {
    string1 = "[img=item/express-transport-belt] String A",
    string2 = "[img=item/express-transport-belt] String A"
  },
  mod_name = "sender",
  name = 145,
  tick = 300
}
 639.718 Script @__sender__/control.lua:19: Data, before event raised:
{
  string1 = "[img=item/express-transport-belt] A",
  string2 = "[img=item/express-transport-belt] B"
}
 639.718 Script @__receiver__/control.lua:3: Event received:
{
  data = {
    string1 = "[img=item/express-transport-belt] A",
    string2 = "[img=item/express-transport-belt] B"
  },
  mod_name = "sender",
  name = 145,
  tick = 300
}
```
Funny how the first example breaks and the almost identical second example is fine.

**Comment by eduran84** [comment]
Klonan asked me to create a bug report: https://forums.factorio.com/viewtopic.php?f=7&t=68801

**Comment by eduran84** [comment]
I narrowed the cause down further. Multiple strings above 40 characters within the same event data table will all be overwritten with one of the long strings. It may or may not be fixed in 0.17.24. I think it is safe to say LTN is not at fault and the issue here can be closed.
